### PR TITLE
2181 redisson enforce marshalling codec

### PR DESCRIPTION
**Pull request by bedaka**
described in issue: https://github.com/factcast/factcast/issues/2181

### CODE REVIEW

**Comment by uweschaefer** [comment]
thx @bedaka , however, i sense a test is missing here :laughing:

File: factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java [hunk 1]
```diff
@@ -49,12 +51,16 @@ public RedissonSnapshotCache(@NonNull RedissonClient redisson, int retentionTime
   @Override
   public @NonNull Optional<Snapshot> getSnapshot(@NonNull SnapshotId id) {
     String key = createKeyFor(id);
-    RBucket<Snapshot> bucket = redisson.getBucket(key);
+    // From redisson-spring-boot-starter 3.19.0 onwards the default codec is Kryo5.
+    // Since it also uses Java17 we have to stick with 3.18.1 and enforce the old default codec
```
Comment (uweschaefer) [inline factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java:55]: the java17 thesis is probably not true.

Comment (bedaka) [reply factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java:55]: I tried but also failed to reproduce it.

File: factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java [hunk 2]
```diff
@@ -28,43 +28,57 @@
...
   private static final String TS_INDEX = "SNAPCACHE_IDX";
   private final RMap<String, Long> index;
 
-  public RedissonSnapshotCache(@NonNull RedissonClient redisson, int retentionTimeInDays) {
+  public RedissonSnapshotCache(
+      @NonNull RedissonClient redisson, @NonNull RedissonSnapshotProperties props) {
     this.redisson = redisson;
-    this.retentionTimeInDays = retentionTimeInDays;
+    this.properties = props;
     index = redisson.getMap(TS_INDEX);
```
Comment (uweschaefer) [inline factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java:52]: shouldn't the codec be also used for the index?

File: factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java [hunk 3]
```diff
@@ -28,43 +28,57 @@
...
 
+  private RBucket<Snapshot> createFromCodec(String key, Codec codec) {
+    return redisson.getBucket(key, codec);
+  }
+
   @Override
   public void setSnapshot(@NonNull Snapshot snapshot) {
     String key = createKeyFor(snapshot.id());
-    redisson.getBucket(key).set(snapshot, retentionTimeInDays, TimeUnit.DAYS);
+    redisson.getBucket(key).set(snapshot, properties.getRetentionTime(), TimeUnit.DAYS);
```
Comment (uweschaefer) [inline factcast-snapshotcache-redisson/src/main/java/org/factcast/core/snap/redisson/RedissonSnapshotCache.java:81]: codec must be used when writing into the bucket

File: factcast-snapshotcache-redisson/src/test/java/org/factcast/core/snap/redisson/RedissonSnapshotCacheTest.java [hunk 4]
```diff
@@ -166,4 +172,19 @@ private void sleep(long millis) {
       Thread.sleep(millis);
     }
   }
+
```
Comment (uweschaefer) [inline factcast-snapshotcache-redisson/src/test/java/org/factcast/core/snap/redisson/RedissonSnapshotCacheTest.java:175]: There are tests missing that validate the use of the codec when reading & writing

**Review by uweschaefer — CHANGES_REQUESTED** [review]

File: factcast-snapshotcache-redisson/src/test/java/org/factcast/core/snap/redisson/RedissonSnapshotCacheTest.java (hunk 4 above)
Comment (bedaka) [reply factcast-snapshotcache-redisson/src/test/java/org/factcast/core/snap/redisson/RedissonSnapshotCacheTest.java:170]: I added one test that checks both. Not sure if it is sufficient though.

**Review by uweschaefer — APPROVED** [review]

**Comment by uweschaefer** [comment]
thx @bedaka

**Pull request merged**